rules/aip0215/versioned_packages.go

```diff
@@ -33,14 +33,16 @@ var versionedPackages = &lint.FileRule{
 		}
 
 		// Ignore this if there is no package.
-		p := strings.Split(f.GetPackage(), ".")
-		if len(p) == 1 && p[0] == "" {
+		segments := strings.Split(f.GetPackage(), ".")
+		if len(segments) == 1 && segments[0] == "" {
 			return false
 		}
 
-		// Exempt anything ending in .type, or .v1master, .v2master, .master, etc.
-		if last := p[len(p)-1]; last == "type" || strings.HasSuffix(last, "master") || strings.HasSuffix(last, "main") {
-			return false
+		// Exempt anything containing .type, or .v1master, .v2master, .master, etc.
+		for _, segment := range segments {
+			if segment == "type" || strings.HasSuffix(segment, "master") || strings.HasSuffix(segment, "main") {
```

Comment:
This seems to significantly relax the rule. 
The original rule excluded checking when the last element was main or master. 
In your new schema any segment with suffix master will cause for the rule to be ignored.

Reply:
Based on the PR description, I believe this was the intent. Am I incorrect?

Reply:
This is indeed the intent. `master` and `main` are the "template" words used by Google's proto versioning tools. They could technically appear at any point in the proto package, just like a version could e.g. `google.foo.bar.baz.v1main.biz.buzz` or `google.foo.master.bar.biz.buz.baz`.